### DISCUSSION TITLE
Problems trying to translate offline page

### DISCUSSION THREAD

**Discussion by juancariza03** [serwist/serwist#44] [Q&A]
First off, thanks @DuCanhGH for your hard work. 
I am trying to implement Serwist into my project, and I'm having an issue. The problem arises when I initially open the website in a specific locale (let's assume I opened it at /en https://... .app/en), it works fine as expected when I navigate on the website. When I go offline and force the page to go to the offline version, the translations display correctly in English as expected. However, when I go back online and switch the locale (let's assume I switched it to /es https://... .app/es), the translations remain in English.

Here is a video to illustrate the problem that I'm having.
[https://youtu.be/V4Qo5FqbXPs](url)
Also here is the repository to replicate the issue.
[https://github.com/juancariza03/pwa-intl](url)

Thank you in advance!
Juan

**Comment by DuCanhGH** [comment]
Seems that you didn't describe the problem you were facing correctly. Only by watching the video did I realize what your problem was.

This was the root cause:
```js
installSerwist({
  // Other options
  fallbacks: {
    entries: [
      {
        url: "/offline",
        revision,
        matcher({ request }) {
          return request.destination === "document";
        }
      }
    ]
  },
});
```
So, with this option, Serwist tried to cache "/offline" on installation. Clearly, the middleware redirected it to "/en/offline", since the language used at that moment was English. As a result, the "/offline" Serwist had prepared was the English one, so any offline request would be redirected to that. To fix this, you should add both offline pages to the `fallbacks.entries` option:
```js
installSerwist({
  // Other options
  fallbacks: {
    entries: [
      {
        url: "/en/offline",
        revision,
        matcher({ request }) {
          // Implement the `getRequestLang` function yourself!
          return request.destination === "document" && getRequestLang(request.url) === "en";
        }
      },
      {
        url: "/es/offline",
        revision,
        matcher({ request }) {
          // Implement the `getRequestLang` function yourself!
          return request.destination === "document" && getRequestLang(request.url) === "es";
        }
      }
    ]
  },
});
```
Hope this helps!
